hyperbolic_agentkit_core/actions/ssh_manager.py

```diff
@@ -58,9 +88,95 @@ def connect(self, host: str, username: str, password: Optional[str] = None,
             self._connected = False
             return f"SSH Connection Error: {str(e)}"
 
-    def execute(self, command: str, interactive: bool = False, questions_and_responses: dict = None) -> str:
+    def interactive_command(
+        self, command: str = "", send_str: str = "", timeout: int = 30
+    ) -> Tuple[str, str]:
+        """
+        Executes a command or sends input in interactive mode and returns the output.
+
+        This method maintains an interactive SSH session that can be used for multiple
+        interactions. It either starts a new command or continues an existing session
+        by sending new input. All output processing and decision making is left to
+        the calling function.
+
+        Args:
+            command: Initial command to execute (if starting a new interaction)
+            send_str: String to send to the session (for continuing interactions)
+            timeout: Maximum time to wait for output
+
+        Returns:
+            Tuple[str, str]: (command_output, error_message)
+            - command_output contains all output received during this interaction
+            - error_message contains any error that occurred, empty string if successful
+        """
+        if not self.is_connected:
+            return "", "Error: No active SSH connection. Please connect first."
+
+        try:
+            # Create the interactive channel if it doesn't exist
+            if (
+                not hasattr(self, "_interactive_channel")
+                or self._interactive_channel is None
+            ):
+                self._interactive_channel = self._ssh_client.invoke_shell()
+                self._interactive_channel.settimeout(timeout)
+                self._interactive_channel.set_combine_stderr(True)
+
+            # Initialize output collection
+            output_buffer = []
+
+            # Send the command or input if provided
+            if command:
+                self._interactive_channel.send((command + "\n").encode("utf-8"))
+            elif send_str:
+                self._interactive_channel.send((send_str + "\n").encode("utf-8"))
+
+            # Wait for and collect output
+            # We'll implement a small delay to allow output to accumulate
+            time.sleep(0.5)  # Give the command some time to generate output
+
+            while True:
+                if self._interactive_channel.recv_ready():
+                    chunk = self._interactive_channel.recv(1024).decode(
+                        "utf-8", "ignore"
+                    )
+                    output_buffer.append(chunk)
+                    # Small sleep to allow more output to arrive
+                    time.sleep(0.1)
```

Comment:
what if it takes more than 0.1 sec to receive the output?

Reply:
hmm it should get caught on the next iteration. This was added to not aggressively pole the channel